src/04-modality-specific-files/01-magnetic-resonance-imaging-data.md

```diff
@@ -161,6 +396,75 @@ can also be used to make that distinction. At what level of detail to make the
 distinction (e.g. just between RARE and FLASH, or between RARE, FLASH, and
 FLASHsubsampled) remains at the discretion of the researcher.
 
+#### The `echo` entity 
+
+If the value of `EchoTime` metadata field varies at least once across a collection 
+of anatomical images having a common `grouping suffix`, the use of `echo-<index>`
+key/value pair is REQUIRED. Note that only integers (from 1 to N) are allowed as
+values to this entity for N different `EchoTime` parameter values. The actual 
+`EchoTime` parameter values MUST NOT be explicitly declared by the entity. Instead, 
+the parameter values are stored in sidecar json files and indexed by the `echo` entity
+in ascending order. For example: 
```

Comment:
I believe that this would be backwards-incompatible (please see [this Google Group thread](https://groups.google.com/g/bids-discussion/c/Utv29uL0Bdc?pli=1)).

Reply:
Shall we make at least make it a recommendation?

Reply:
I think that would be great!

Reply:
@Gilles86 by recommendation, do you mean the following edit: 

```diff
Instead, the parameter values are stored in sidecar json files and 
+ RECOMMENDED 
to be indexed by the `echo` entity in ascending order. For example: 
```

Reply:
Yes, this is something I added to the BIDS 2.0 doc as `REQUIRED`. The best we can do right now is recommend !

Reply:
There is now an issue about making ascending `echo` ordering required for BIDS 2.0: https://github.com/bids-standard/bids-2-devel/issues/40.